service/storage_proxy.cc

```diff
@@ -1750,6 +1750,31 @@ storage_proxy::create_write_response_handler_helper(schema_ptr s, const dht::tok
     std::vector<gms::inet_address> natural_endpoints = rs.get_natural_endpoints(token);
     std::vector<gms::inet_address> pending_endpoints = _token_metadata.pending_endpoints_for(token, keyspace_name);
 
+    if (get_local_storage_service().get_token_metadata().is_any_node_being_replaced() &&
+        rs.get_type() != locator::replication_strategy_type::local) {
```

Comment:
how is this different from any user keyspace with RF=1 ?

Reply:
With local strategy, the replacing node will not be in the pending_endpoints. 
With other strategy, the replacing node will be in the pending_endpoints. 

If we filter the replacing node out of natural_endpoints for local strategy, no node will be found for the write.

Reply:
Well, it is nicer to hide it in abstract_replication_strategy. But let's have a basic version that works and we agree on, then go for the nicer thing.

Reply:
Agree with tgrabiec.